core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -93,10 +96,22 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
         return hashCode();
       } else if ("toString".equals(method.getName())) {
         return toString();
+      } else if (Util.isDefault(method)) {
+        return invokeDefault(proxy,method, args);
       }
       return dispatch.get(method).invoke(args);
     }
 
+    private Object invokeDefault(Object proxy, Method method, Object[] args) throws Throwable {
```

Comment:
Since this is guarded by Util.isDefault, we can assume that the language level isn't 6 when we call this.

You should be able to get by with `@IgnoreJRERequirement  // Should only be invoked on Java 7+.` on this method